### PR TITLE
Add google authentication feature

### PR DESCRIPTION
**Pull request by Jigao-Zeng**
---
name: Add google authentication 
about: Google authentication 
title: 'Enable google authentication '
labels: issue #175 
assignees:  Jigao Zeng
---

**Describe the issue**
The user can't sign in using google authentication, which is not convenient if they forgot their account details.

**Describe the solution**
Add google sign in based on firebase.

### CODE REVIEW

File: src/pages/loginView.js [hunk 1]
```diff
@@ -58,7 +58,21 @@ export default function LoginView() {
   };
 
   /**
-   * Reset password
+   * handle google authentication
+   */
+  const googleLoginHandler = async () => {
```
Comment (dikunong) [inline src/pages/loginView.js:63]: This code should be moved into `authService.js`, where all existing Firebase authentication stuff lives.

**Review by dikunong — CHANGES_REQUESTED** [review]
Good job so far. To maintain separation of concerns, we shouldn't be mixing frontend with Firebase code. As such, I would suggest moving `googleLoginHandler` out the frontend and into `authService.js` so authentication functionality remains in one place.

Also, has this been tested with an actual Google account? We store first/last names in the users database, which doesn't intersect with the auth system. I can't see the first and last names being set anywhere - this means that users would have a blank name upon first run, which isn't a great experience.  
I suggest taking a look at the existing `signUp` method on line 34 of `authService.js` - they demonstrate persisting names to the users database there. Please clarify on this and correct me if I'm wrong :)

File: src/pages/loginView.js (hunk 1 above)
Comment (Jigao-Zeng) [reply src/pages/loginView.js:63]: Thanks for your review, I will take a look at how sign up is worked and adjust my function.

**Review by Denijar — CHANGES_REQUESTED** [review]
Hi Jigao - after clicking Google Login and trying to login with a google account, I got this error and I am now unable to use the application, even after refreshing.
image

**Comment by Jigao-Zeng** [comment]
I think the reason is the user didn't sign up first before using google log in, so no information is stored. My idea is to do a "sign up" first if the user haven't by storing their information, and then load the home page.

**Comment by Denijar** [comment]
So after I got this error, even when I switched to `main-group2` and ran the application, I still got this error which was very very scary. For now I have disabled Google Authentication in Firebase. Given how late this PR was put up and the extent of the error that it caused, I am very hesitant to allow the PR to go through for the deadline tonight.

**Comment by Jigao-Zeng** [comment]
Yea, I am fixing the big problem. Can I ask user to go back to sign up page if their google email is not being registered?

**Pull request closed without merging**